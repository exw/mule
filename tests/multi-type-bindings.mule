{
  , type t = int
  , type u = t
}
