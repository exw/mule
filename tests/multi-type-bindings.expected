
Desugared:

	(App
	 ((Fix Record)
	  (Lam
	   ($0
	    (LetType
	     (((t (Recur (() t (Var (() int))))) (u (Recur (() u (Var (() t))))))
	      (App
	       ((App
	         ((Update (Type t))
	          (App
	           ((App ((Update (Type u)) EmptyRecord)) (Witness (Var (() u)))))))
	        (Witness (Var (() t)))))))))))

inferred type:

	(Quant
	 (() All a Unknown
	  (Record (r_info ())
	   (r_types (() ((t (Named (() int))) (u (Named (() int)))) (a)))
	   (r_values (() () ())))))

Runtime term:

	(App ((Fix Record) (Lam (0 () (Record ())))))

Evaluated:

	(Record ())
